Chris git branch

chrisdemars opened the pull request:
I would like my changes to be merged in please.

levibostian commented on git.html.markdown:
```diff
@@ -238,7 +241,7 @@ $ git clone -b master-cn https://github.com/adambard/learnxinyminutes-docs.git -
 
 Stores the current contents of the index in a new "commit." This commit contains
-the changes made and a message created by the user.
+the changes made and a message created by the user. *If you do not add the "-m" flag, git will throw you in to Vim or Nano, and your world will explode.*
```
Would be better if we said, `*If you do not add the '-m' flag, git will open up a text editor for you to enter in the commit message.` I have mine set for emacs for it is neither vim or nano...and my world does not explode, emacs makes me happy :)

chrisdemars replied:
Ok, I will change that. The only reason I added the "world will explode"
comment is because I taught a workshop on using git and most users that
don't use it heavily forget to put the -m in and they are thrown in to vim
or nano, not even having a text editor specified as the default and they
freak out and don't know what to do and start smashing buttons.

On Wed, Oct 14, 2015 at 10:30 AM, Levi Bostian <notifications@github.com>
wrote:

> In git.html.markdown
> <https://github.com/adambard/learnxinyminutes-docs/pull/1487#discussion_r42000429>
> :
>
> > @@ -238,7 +241,7 @@ $ git clone -b master-cn https://github.com/adambard/learnxinyminutes-docs.git -
> >  ### commit
> >
> >  Stores the current contents of the index in a new "commit." This commit contains
> > -the changes made and a message created by the user.
> > +the changes made and a message created by the user. *If you do not add the "-m" flag, git will throw you in to Vim or Nano, and your world will explode.*
>
> Would be better if we said, *If you do not add the '-m' flag, git will
> open up a text editor for you to enter in the commit message. I have mine
> set for emacs for it is neither vim or nano...and my world does not
> explode, emacs makes me happy :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adambard/learnxinyminutes-docs/pull/1487/files#r42000429>
> .
>

levibostian replied:
Haha is that right? I am pretty sure that happened to me the first time that happened to me too.

chrisdemars replied:
It did, did u want me to merge in to master on my local or were you going
to merge my branch in to master?

On Thu, Oct 15, 2015 at 2:35 PM, Levi Bostian <notifications@github.com>
wrote:

> In git.html.markdown
> <https://github.com/adambard/learnxinyminutes-docs/pull/1487#discussion_r42162522>
> :
>
> > @@ -238,7 +241,7 @@ $ git clone -b master-cn https://github.com/adambard/learnxinyminutes-docs.git -
> >  ### commit
> >
> >  Stores the current contents of the index in a new "commit." This commit contains
> > -the changes made and a message created by the user.
> > +the changes made and a message created by the user. *If you do not add the "-m" flag, git will throw you in to Vim or Nano, and your world will explode.*
>
> Haha is that right? I am pretty sure that happened to me the first time
> that happened to me too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adambard/learnxinyminutes-docs/pull/1487/files#r42162522>
> .
>

chrisdemars commented:
Changed the copy for your review.

levibostian commented:
Awesome, thanks. 

Can you merge in master please?

chrisdemars commented:
You want me to merge my branch in to master?

vendethiel commented:
The opposite

chrisdemars commented:
@vendethiel what do you mean? Merge my branch in to master on my local repo then submit another PR?

levibostian commented:
@chrisdemars there are merge conflicts issues right now with this pull request so we are not able to merge it. 

So on your local machine, do the following commands:
```
git fetch upstream
git merge upstream/master
```
Then, you will see `CONFLICT` messages for files with conflicts. Go into those files and fix the conflicts. 

Then do a commit and push. 
This will update this PR and we will be able to merge it.

onebree commented on git.html.markdown:
```diff
@@ -238,7 +241,7 @@ $ git clone -b master-cn https://github.com/adambard/learnxinyminutes-docs.git -
 
 Stores the current contents of the index in a new "commit." This commit contains
-the changes made and a message created by the user.
+the changes made and a message created by the user. *If you do not add the "-m" flag, git will open up a text editor for you to enter your commit message.*
```
Just a small change, but change "-m" to `-m`.

Bigger change, word it as:
> By default, git will open up a text editor to enter a commit message. To write the message in-line, use the flag `-m`.

chrisdemars commented:
@levibostian I am trying to do the git fetch upstream and I can't, I tried merging my branch in to master and it says its up to date. What am I doing wrong?

onebree commented:
@Chris54721 I suggest the following:
1. Create a remote for this repo (`remote add adambard <url>`)
2. `git fetch adambard`
3. While on your feature branch, `git merge --no-commit adambard/master`
4. You'll be prompted with conflicts. Fix them, then commit to your branch, and push

chrisdemars commented:
@onebree I did all that and it says there are conflicts, but it is not showing me the conflicts.

onebree commented:
@Chris54721 when you do `git status`, are any files shown as "both modified", or something?

chrisdemars commented:
@onebree it shows a bunch of files modified and one says both modified. And also, when you are mentioning me, that isnt my user name but I keep getting the notifications lol.

onebree commented:
You get notified because it is an issue you subscribed to (since you authored it). Open the one that says "both modified" and you should see something like:

```
<<<< CHRIS-BRANCH
CODE CODE CODE
CODE CODE CODE
====
CODE1 CODE1 CODE1
>>>> ADAM/MASTER
```

chrisdemars commented:
@levibostian I think I fixed the code and commited the changes. What do I have to do now for them to merged?

@onebree I seen that in the code. I fixed it, I think lol. I added and commited the file. Now I just need it to be merged, I think.

onebree commented:
@chrisdemars looks like everything is good! Github says there are no conflicts, so your branch can be merged. However, only maintainers of this repo can merge it... probably after they review your changes to make sure the conflicts were resolved correctly.

chrisdemars commented:
@onebree right right. This is the first time I have actively contributed to an OS project so this all new to me. Working locally with your own files and projects is one thing, but working with OS stuff is a different beast lol.

chrisdemars commented:
Did my changes ever get merged?

onebree commented:
@chrisdemars You will know whether your changes are merged because this pull request will say "merged" in a purple box. (Rather than green everywhere)

chrisdemars commented:
@onebree from what I can tell it still says open.

onebree commented:
@chrisdemars Yes it still is. Let me ping someone so they can maybe review your changes. Please be patient.

@levibostian

chrisdemars commented:
Ah ok. Thanks @onebree

vendethiel commented:
sorry for the delay, it seems there are merge conflicts now :(.

vendethiel commented:
Going to close this one for now :). Feel free to ping us here if you find some time working on it again! Thanks.
